crates/conjure_core/src/rules/base.rs

```diff
@@ -77,11 +77,16 @@ fn remove_nothings(expr: &Expr, _: &Model) -> ApplicationResult {
     }
 }
 
-/// Remove empty expressions:
-///  ```text
-///   or([]) ~> false
-///   X([])  ~> Nothing
-///  ```
+/// This rule simplifies expressions where the operator is applied to an empty set of sub-expressions.
+///
+/// For example:
+/// - `or([])` simplifies to `false` since no disjunction exists.
+///
+/// **Applicable examples:**
+/// ```text
+/// or([])  ~> false
+/// X([]) ~> Nothing
+/// ```
 #[register_rule(("Base", 8800))]
 fn remove_empty_expression(expr: &Expr, _: &Model) -> ApplicationResult {
```

Comment:
I just noticed: why is this a separate rule instead of being handled by the full evaluator?

Reply:
@niklasdewally @YehorBoiar @lixitrixi

Reply:
By the look of git blame, seems to just be because Nothing was added before the evaluator.

 I agree that it should be an evaluation step.

Reply:
bruh

Reply:
don't know what you mean by "We should just define an empty expression behaviour in rules"

for context, full evaluator here: https://github.com/conjure-cp/conjure-oxide/blob/dc7f98bd5cdf32122de3ce4fb8235d86747d4b88/crates/conjure_core/src/rules/constant.rs#L11

partial evaluator here: https://github.com/conjure-cp/conjure-oxide/blob/dc7f98bd5cdf32122de3ce4fb8235d86747d4b88/crates/conjure_core/src/rules/partial_eval.rs

Reply:
and if it all was working correctly this would already do it? https://github.com/conjure-cp/conjure-oxide/blob/dc7f98bd5cdf32122de3ce4fb8235d86747d4b88/crates/conjure_core/src/rules/constant.rs#L42

I think there is a bug. how well tested is the full evaluator?

remove_empty_expression is at 8800, and full evaluator is at 9001, so that should take precedence even if this rule was buggy (which I think it is)